src/robusta/integrations/kubernetes/custom_models.py

```diff
@@ -390,8 +395,30 @@ def get_single_pod(self) -> RobustaPod:
             raise Exception(f"got more pods than expected for job: {pods}")
         return pods[0]
 
+    def create_job_owned_secret(self, secret: JobSecret):
+        """
+            This secret will be auto-deleted when the pod is Terminated
+        """
+        # Due to inconsistant GC in K8s the OwnerReference needs to be the pod and not the job (Found in azure)
+        job_pod = self.get_single_pod()
+        robusta_owner_reference = OwnerReference(apiVersion="v1",
+                                                 kind="Pod",
+                                                 name=job_pod.metadata.name,
+                                                 uid=job_pod.metadata.uid,
+                                                 blockOwnerDeletion=False,
+                                                 controller=True)
+        secret = Secret(
+            metadata=ObjectMeta(name=secret.name, ownerReferences=[robusta_owner_reference]),
+            data=secret.data
+        )
+        try:
+            return secret.createNamespacedSecret(job_pod.metadata.namespace).obj
+        except Exception as e:
+            logging.error(f"Failed to create secret {secret.name}", exc_info=True)
```

Comment:
Secret here is the k8s secret object, it doesn't have name
I think you have ambiguous names

Reply:
youre right